types.d.ts

```diff
@@ -1,360 +1,370 @@
-
-      /**
-       * Property representing the actual value to set the flag to.
-       *
-       * It is a string, that may be converted to the right type
-       * (that is `true` or `false` for a `boolean` value, or `4` for a `number` value).
-       *
-       * It doesn't need to be enclosed in `<code>` tags.
-       */
-      value_to_set?: string;
-    }[];
-    /**
-     * A `boolean` value indicating whether or not the implementation of the sub-feature follows
-     * the current specification closely enough to not create major interoperability problems.
+     * Property representing the actual value to set the flag to.
      *
-     * It defaults to `false` (no interoperability problem expected).
+     * It is a string, that may be converted to the right type
+     * (that is `true` or `false` for a `boolean` value, or `4` for a `number` value).
      *
-     * If set to `true`, it is recommended to add a note indicating how it diverges from
-     * the standard (implements an old version of the standard, for example).
+     * It doesn't need to be enclosed in `<code>` tags.
      */
-    partial_implementation?: boolean;
+    value_to_set?: string;
+  }[];
+  /**
+   * A `boolean` value indicating whether or not the implementation of the sub-feature follows
+   * the current specification closely enough to not create major interoperability problems.
+   *
+   * It defaults to `false` (no interoperability problem expected).
+   *
+   * If set to `true`, it is recommended to add a note indicating how it diverges from
+   * the standard (implements an old version of the standard, for example).
+   */
+  partial_implementation?: boolean;
 
-    /**
-     * An `array` of zero or more strings containing
-     * additional information. If there is only one entry in the array,
-     * the array must be omitted.
-     *
-     * The `<code>` and `<a>` HTML elements can be used.
-     */
-    notes?: string | string[];
-  }
+  /**
+   * An `array` of zero or more strings containing
+   * additional information. If there is only one entry in the array,
+   * the array must be omitted.
+   *
+   * The `<code>` and `<a>` HTML elements can be used.
+   */
+  notes?: string | string[];
+}
 
-  interface Identifier
-    extends Partial<Record<Exclude<string, '__compat'>, Identifier>> {
-    /**
-     * A feature is described by an identifier containing the `__compat` property.
-     *
-     * In other words, identifiers without `__compat` aren't necessarily features,
-     * but help to nest the features properly.
-     *
-     * When an identifier has a `__compat` block, it represents its basic support,
-     * indicating that a minimal implementation of a functionality is included.
-     *
-     * What it represents exactly depends of the evolution of the feature over time,
-     * both in terms of specifications and of browser support.
-     */
-    __compat?: CompatStatement;
-  }
+export type Identifier = PrimaryIdentifier & IdentifierMeta;
```

Comment:
Indentifier references Primary Identifier but then it references it back... ?

Reply:
It works fine.

This is because I can’t define `__compat?: CompatStatement` in an interface which extends `Record<string, Identifier>`.